flamedisx/block_source.py

```diff
@@ -144,6 +147,117 @@ def annotate(self, d: pd.DataFrame):
                           <= d[f'{dim}_max'].values), \
                 f"_annotate of {self} set misordered bounds"
 
+    def bayes_bounds_binomial(self, df, in_dim, supports, rvs_binom, ns_binom, ps_binom, bound):
+        """Calculate bounds on a block using a binomial distribution.
+
+        :param df: Dataframe with events
+        :param in_dim: String giving the 'input' dimension to the block, whose bounds are
+        being calculated
+        :param supports: Values of in_dim over which the PMF/CMF used to find the bounds
+        will be calculated, for each event in the dataframe
+        :param rvs_binom: Variable the block uses as the 'object' of the binomial calculation;
+        must be the same shape as supports
+        :param ns_binom: Variable the block uses as the number of trials of the binomial calculation;
+        must be the same shape as supports
+        :param ps_binom: Variable the block uses as the success probability of the binomial calculation;
+        must be the same shape as supports
+        :param bound: 'upper', 'lower' or 'mle' to determine which bound is currently being calculated
+        """
+        assert (bound == 'upper' or 'lower' or 'mle'), "bound argumment must be upper, lower or mle"
+        assert (np.shape(rvs_binom) == np.shape(ns_binom) == np.shape(ps_binom) == np.shape(supports)), \
+            "Shapes of suports, rvs_binom, ns_binom and ps_binom must be equal"
+
+        def binomial(x, n, p):
```

Comment:
Why not use scipy.stats' functions for this and other basic distribution calculations below? E.g. `stats.binom(n=.., p=..).pmf(k)`. These are fast, vectorized, extensively tested, and they will even use appropriate approximations when possible. Flamedisx also depends on scipy already.

Reply:
Done, tested, all good

Reply:
Great, the stuff in bounds.py looks a lot cleaner/clearer now.